src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -124,7 +124,7 @@ private BoundExpression RewriteDeconstruction(BoundTupleExpression left, Convers
             {
                 rightParts = ((BoundTupleExpression)right).Arguments;
             }
-            else if (right.Kind == BoundKind.Conversion)
+            else if (right.Kind == BoundKind.Conversion && ((BoundConversion)right).Operand.Kind == BoundKind.ConvertedTupleLiteral)
```

Comment:
>((BoundConversion)right).Operand.Kind == BoundKind.ConvertedTupleLiteral [](start = 59, length = 72)

Can we by any chance get here with TupleLiteral? Would it be more robust to do a type check for BoundTupleExpression instead?